### PR TITLE
Improve TM testing support infrastructure

### PR DESCRIPTION
**Pull request by Didainius**
* Improve scripts for TM test support - `make testtm-acc`, `testtm-acc-coverage`, `make testtm-binary`, `make
  testtm-binary-prepare`
* Adjusts `TESTING.md` for TM testing
* Splits sample test configs to `sample_vcd_test_config.json` for VCD and `sample_vcd_test_config_tm.json` for TM

### CODE REVIEW

File: vcd/terraform_binary_test.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-//go:build binary
+//go:build binary && !tm
```
Comment (Didainius) [inline vcd/terraform_binary_test.go:1]: The test templates files that are in `vcd/test-templates` are not compatible with TM therefore this is not being executed when `tm` tag is specified

File: scripts/runtest.sh [hunk 2]
```diff
@@ -99,11 +99,11 @@ function short_test {
     fi
     if [ -n "$VERBOSE" ]
     then
-        echo "VCD_SHORT_TEST=1 go test -race -tags 'functional $MORE_TAGS' -v -timeout 5m"
+        echo "VCD_SHORT_TEST=1 go test -race -tags '$MORE_TAGS' -v -timeout 5m"
```
Comment (Didainius) [inline scripts/runtest.sh:102]: `functional` tag was default until TM, but this tag is now moved upstream to the function caller and it should be included within `$MORE_TAGS` in the caller. This allows for isolating it

**Review by adambarreiro — COMMENTED** [review]
LGTM, will give it a try and then approve

File: TESTING.md [hunk 3]
```diff
@@ -147,14 +154,16 @@ In order to test the provider, you can simply run `make test`.
...
-In order to run the full suite of Acceptance tests, run `make testacc`.
+In order to run the full suite of Acceptance tests for **Cloud Director**, run `make testacc`.
 
 *Note:* Acceptance tests create real resources, and often cost money to run.
 
 ```sh
 $ make testacc
 ```
 
+To run full suite of **Tenant Manager** acceptance tests, run `make testtmacc`.
```
Comment (lvirbalas) [inline TESTING.md:165]:
```suggestion
To run full suite of **Tenant Manager** acceptance tests, run `make testtm-acc`.
```

File: TESTING.md [hunk 4]
```diff
@@ -311,18 +320,26 @@ terraform tool through a shell script, and for every test we run
...
+```bash
+make testtm-binary
+```
+
 All the tests run unattended, stopping only if there is an error.
 
-It is possible to customise running of the binary tests by preparing them and then running the test script from the `tests-artifacts` directory:
+It is possible to customise running of the binary tests by preparing them and then running the test
+script from the `tests-artifacts` directory. For **Cloud Director** the first command is `make
+test-binary-prepare`, while for **Tenant Manager**, it is `testtm-binary-prepare`:
```
Comment (lvirbalas) [inline TESTING.md:339]: I'm suggesting to split this into two pieces and then remove paragraph slightly below. Would look something like this:

> script from the `tests-artifacts` directory. For **Cloud Director** the first command is:
> 
> ```make
> test-binary-prepare
> ```
> 
> while for **Tenant Manager** it is:
> 
> ```make
> testtm-binary-prepare
> ```
>

File: TESTING.md [hunk 5]
```diff
@@ -311,18 +320,26 @@ terraform tool through a shell script, and for every test we run
...
 
 ```bash
-make test-binary-prepare
+make test-binary-prepare # or make testtm-binary-prepare for Tenant Manager
```
Comment (lvirbalas) [inline TESTING.md:342]: With above change, we can then remove this paragraph completely.

File: TESTING.md (hunk 3 above)
Comment (Didainius) [reply TESTING.md:165]: Fixed.

File: TESTING.md (hunk 4 above)
Comment (Didainius) [reply TESTING.md:339]: Changed.

File: TESTING.md (hunk 5 above)
Comment (Didainius) [reply TESTING.md:342]: Changed.

**Pull request merged**